src/components/Slider.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, createElement, DOM } from "react";
+
+import * as classNames from "classnames";
+import * as RcSlider from "rc-slider";
+
+import "rc-slider/dist/rc-slider.css";
+import "../ui/Slider.css";
+
+import { Alert } from "./Alert";
+
+export interface SliderProps {
+    value?: number;
+    noOfMarkers?: number;
+    maxValue?: number;
+    minValue?: number;
+    alertMessage?: string;
+    onChange?: (value: number) => void;
+    onUpdate?: (value: number) => void;
+    stepValue?: number;
+    tooltipText?: string | null;
+    disabled: boolean;
+    decimalPlaces?: number;
+}
+
+interface Marks {
+    [key: number]: string | number | {
+        style: HTMLStyleElement,
+        label: string
+    };
+}
+
+export class Slider extends Component<SliderProps, {}> {
+    static defaultProps: SliderProps = {
+        disabled: false,
+        maxValue: 100,
+        minValue: 0,
+        noOfMarkers: 2,
+        tooltipText: "{1}"
+    };
+
+    constructor(props: SliderProps) {
+        super(props);
+
+        this.getTooltipText = this.getTooltipText.bind(this);
+    }
+
+    render() {
+        const { alertMessage } = this.props;
+        return DOM.div({ className: classNames("widget-slider", { "has-error": !!alertMessage }) },
+            createElement(RcSlider, {
+                disabled: this.props.disabled,
+                included: true,
+                marks: this.calculateMarks(this.props),
+                max: this.props.maxValue,
+                min: this.props.minValue,
+                onAfterChange: this.props.onChange,
+                onChange: this.props.onUpdate,
+                pushable: false,
+                range: false,
+                step: this.props.stepValue,
+                tipFormatter: this.props.tooltipText ? this.getTooltipText : null,
+                value: typeof this.props.value === "number" ? this.props.value : this.calculateDefaultValue(this.props)
+            }),
+            alertMessage && !this.props.disabled ? createElement(Alert, { message: alertMessage }) : null
+        );
+    }
+
+    private calculateMarks(props: SliderProps): Marks {
+        const marks: Marks = {};
+        const { noOfMarkers, maxValue, minValue } = props;
+        if ((noOfMarkers || noOfMarkers === 0) && (maxValue || maxValue === 0) && (minValue || minValue === 0)) {
```

Comment:
I usually get issues when i do something like this:
const hasMinValue = (minValue || minValue === 0);
if (hasMinValue){
value = minvalue;
}

If there any better way to do it??